knife/lib/chef/knife/bootstrap/train_connector.rb

```diff
@@ -127,8 +127,11 @@ def temp_dir
                         # running with sudo right now - so this directory would be owned by root.
                         # File upload is performed over SCP as the current logged-in user,
                         # so we'll set ownership to ensure that works.
-                        run_command!("chown #{config[:user]} '#{dir}'") if config[:sudo]
-
+                        if config[:sudo]
+                          # While using ssh config file user might present in ssh file
+                          user = config[:user] || ssh_config_for_host(config[:host])[:user]
```

Comment:
This is already available in @config[:user], which we should use instead of re-doing override parsing.

Reply:
@marcparadise, This change will resolve user nil(@config[:user]) issue when we have added user in ssh_config file instead of adding it in bootstrap command.

Reply:
@config contains the merged results of all config, including the user's ssh config as provided by `Net::SSH.config_for_host`.  You can see this in `transport_config`[1], where it also merges in `missing_opts_from_ssh_config` defined later in the same file.  This merge currently accepts `SSH_CONFIG_OVERRIDE_KEYS`, which are `user`, `port`, and `proxy`. 

[1] https://github.com/chef/chef/blob/master/knife/lib/chef/knife/bootstrap/train_connector.rb#L225

Reply:
Thanks @marcparadise, will update according to it.